src/System/Terminal/Render.hs

```diff
@@ -1,14 +1,16 @@
-{-# LANGUAGE UndecidableInstances #-}
 {-# OPTIONS_GHC -Wno-name-shadowing #-}
 
 module System.Terminal.Render where
 
+import Control.Monad.State (MonadState (..))
 import Data.Generics.Product qualified as Lens
 import Data.Text qualified as Text
 import GHC.Records qualified as GHC
 import Internal.Prelude
 import Internal.Prelude qualified as Prelude
 
+-- | Cursor represents the position of the cursor by its row and column
+-- position on the terminal.
```

Comment:
It's important to note here that this is different from `RopeZipper.Position`, which is `(Word, Word)`, because the cursor position is relative to the start of the widget. We explicitly allow negative values here.

Reply:
Noted. I added a comment explaining this nuance